### ISSUE TITLE
Attach associated records during creation 

### ISSUE THREAD

**Issue by Paul-Bob** [avo-hq/avo#3120]
Originally requested here 
https://discord.com/channels/740892036978442260/1261352281287295037/1271485634833285183

During creation the user should be able to attach associated records.

For example if an player has many skills there should be possible to attach skills to the player while creating it.

**Comment by adrianthedev** [comment]
On the New view you can add any fields you want, so maybe this issue is about adding fields when attaching a record?

**Comment by Paul-Bob** [comment]
> On the New view you can add any fields you want

True

> so maybe this issue is about adding fields when attaching a record?

No, that is possible using the [`attach_fields` DSL](https://docs.avohq.io/3.0/associations/has_many.html#attach_fields)

---

This issue concerns pre-selecting records to be attached when a new record is created.

For example, if a `Player` has a `has_many` association with `Skills`, there should be a native way to pre-select skills during the creation of a player, directly within the form. Essentially, the `has_many` field should include an edit component that allows you to pre-select associated records.

@adrianthedev let me know if it’s clearer now or if you have any questions.

---

@julianrubisch could you please confirm that this is your use-case as well but for `has_and_belongs_to_many`?

**Comment by julianrubisch** [comment]
yes, this exactly sums up what I'm trying to do. what's essential to grok the scope of the issue, is that there might be hundreds of `Skills` (well, in this case its a bit stretched, but bear with me), so browsing through a single select box when attaching is just unfeasible. 

it'd be much better to be able to actually filter down the `Skills` somehow, then select a few via checkboxes and hit "Attach" (or skip the checkboxes and say "Attach all")

**Comment by adrianthedev** [comment]
Fixed with the [nested records addon](https://avohq.io/addons/nested-records).

Demo: https://main.avodemo.com/avo/resources/fish/new
